manager/src/main/java/org/openremote/manager/app/ConfigurationResourceImpl.java

```diff
@@ -43,13 +52,71 @@ public ConfigurationResourceImpl(TimerService timerService, ManagerIdentityServi
 
     @Override
     public Object update(RequestParams requestParams, Object managerConfiguration) {
-        this.configurationService.saveMangerConfig(managerConfiguration);
+        this.configurationService.saveManagerConfigFile(managerConfiguration);
         return managerConfiguration;
     }
 
     @Override
     public String fileUpload(RequestParams requestParams, String path, FileInfo fileInfo) {
         this.configurationService.saveImageFile(path, fileInfo);
-        return path;
+        requestParams.getExternalSchemeHostAndPort();
+        URI managerConfigPath = requestParams.getExternalBaseUriBuilder()
+                .path("master")
+                .path("configuration")
+                .path("manager")
+                .path("image")
+                .path(path)
+                .build();
+        return managerConfigPath.toString();
+    }
+
+    @Override
+    public ObjectNode getManagerConfig() {
+        Optional<File> file = configurationService.getManagerConfig();
+        if (file.isEmpty()) {
+            throw new NotFoundException("Configuration file not found");
+        }
+
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode jsonNode = mapper.readTree(file.get());
+            if (jsonNode.isObject()) {
+                ObjectNode objectNode = (ObjectNode) jsonNode;
+                return objectNode;
+            } else {
+                throw new IOException("The provided JSON is not an object.");
+            }
+        } catch (IOException e) {
+            LOG.severe("Error reading manager config file: " + e.getMessage());
+        }
+        return null;
+    }
+
+    @Override
+    public Object getManagerConfigImages(String fileName) {
+
+        try {
+            File imageFile = configurationService.getManagerConfigImage(fileName).orElseThrow();
+            if (!imageFile.exists()) {
+                return Response.status(Response.Status.NOT_FOUND).build();
+            }
+
+            String mimeType = Files.probeContentType(imageFile.toPath());
+            if (mimeType == null) {
+                mimeType = "application/octet-stream";
+            }
+
+            Response.ResponseBuilder response = Response.ok(imageFile, mimeType);
+            response.header("Access-Control-Allow-Origin", "*");
+            response.header("Access-Control-Allow-Methods", "GET, POST, DELETE, PUT");
+            response.header("Access-Control-Allow-Headers", "Content-Type, api_key, Authorization");
```

Comment:
CORS headers should be automatically added and are configured with environment variables, we don't just let any origin make calls to our API